### ISSUE TITLE
[Proposal]Support for padding around `BottomNavigationBarItem.label`

### ISSUE THREAD

**Issue by marctan** [flutter/flutter#104234]
Before in BottomNavigationBar.title we can add style and padding. 

But title is already deprecated and now we need to use `selectedLabelStyle` to style the label. But what if i need to add padding to the label? what should we use? 

Before we can do this: 
```
BottomNavigationBarItem(
    title: Padding(....child: Text(...))
)
```
But i'm not sure how to do it now in label? 

Thanks!

**Comment by maheshmnj** [comment]
Hi @marctan, Thanks for filing the issue. This is currently not possible since the Widget `title` was deprecated and replaced with a String `label`. Labeling this issue as a proposal and for further insights from the team.
More about the deprecation 
https://docs.flutter.dev/release/breaking-changes/bottom-navigation-title-to-label

**Comment by HansMuller** [comment]
Labeled widgets, like Tooltips, NavigationDesitinations, MenuItems, etc,. need a string for the sake of a11y, but would be better off with a widget for the sake of this kind of flexibility. @gspencergoog - we need some grand unification here :-)

**Comment by gspencergoog** [comment]
Yeah, my solution on MenuItem was to require a string (that was the default for a11y), and allow you to also give a widget that would be used in place of the default `Text(label)`.  Not sure it's a good solution, but...

**Comment by HansMuller** [comment]
That sounds reasonable to me. Hopefully we can do it consistently in every widget that needs a label.
